test.js

```diff
@@ -372,6 +399,117 @@ test('Should compress json data (gzip) - global', t => {
   })
 })
 
+test('Should not compress on x-no-compression header', t => {
+  t.plan(2)
+  const fastify = Fastify()
+  fastify.register(compressPlugin, { threshold: 0 })
+  const json = { hello: 'world' }
+
+  fastify.get('/', (req, reply) => {
+    reply.send(json)
+  })
+
+  fastify.inject({
+    url: '/',
+    method: 'GET',
+    headers: {
+      'x-no-compression': true
+    }
+  }, res => {
+    t.strictEqual(res.statusCode, 200)
+    t.notOk(res.headers['content-encoding'])
+  })
+})
+
+test('Should not try compress missing payload', t => {
+  t.plan(2)
+  const fastify = Fastify()
+  fastify.register(compressPlugin, { threshold: 0 })
+  const json = { hello: 'world' }
+
+  fastify.get('/', (req, reply) => {
+    reply.send(undefined)
+  })
+
+  fastify.inject({
+    url: '/',
+    method: 'GET',
+    headers: {
+      'accept-encoding': 'gzip'
+    }
+  }, res => {
+    t.strictEqual(res.statusCode, 200)
+    t.notOk(res.headers['content-encoding'])
+  })
+})
+
+test('Should not compress if content-type is a invalid type', t => {
+  t.plan(2)
+  const fastify = Fastify()
+  fastify.register(compressPlugin, { threshold: 0 })
+  const json = { hello: 'world' }
+
+  fastify.get('/', (req, reply) => {
+    reply.header('content-type', 'something/invalid')
```

Comment:
This test is not needed, `reply.type` is an alias of `reply.header('content-type', ` :)